File: Dataset/Leetcode/train/55/130.cpp
class Solution {
public:
    bool XXX(vector<int>& nums) {
        int pmax=0,cur=0;
        //pmax记录能走到最远的位置，cur用来更新pmax
        while(cur<=pmax && pmax<nums.size()-1){
            pmax=max(pmax,cur+nums[cur]);
            cur++;
        }
        return pmax>=nums.size()-1;
    }
};

File: Dataset/Leetcode/train/46/130.py
class Solution:
    def XXX(self, nums: List[int]) -> List[List[int]]:
        res=[]
        def dfs(l, num):
            if l==len(num):
                print(num)//测试语句
                res.append(num)
                return
            for r in range(l, len(num)):
                num[l], num[r] = num[r], num[l]
                dfs(l+1, num)
                num[l], num[r] = num[r], num[l]
        dfs(0, nums)
        print(res)
        return res

